docs/userguide/CDN/USERGUIDE.md

```diff
@@ -0,0 +1,289 @@
+
+```php
+$services = $cdnService->listServices();
+foreach ($services as $service) {
+    /** @var $service OpenCloud\CDN\Resource\Service **/
+}
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/list-services.php) ]
+
+### Get a service
+
+You can retrieve a specific service by using that service's ID, as shown in the following example:
+
+```php
+$service = $cdnService->getService('0e09ad12-2bfe-4607-80fd-116fa68d9c79');
+/** @var $service OpenCloud\CDN\Resource\Service **/
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/get-service.php) ]
+
+### Update a service
+
+This operation takes one parameter, an associative array, with the following keys:
+
+| Name | Description | Data type | Required? | Default value | Example value |
+| ---- | ----------- | --------- | --------- | ------------- | ------------- |
+| `domains` | An array of associative arrays, each specifying a domain name for your service. | Array of associative arrays | Yes | `null` | `array( array( 'domain' => 'www.acme.com' ), array ( 'domain' => 'acme.com' ) )` |
+| `origins` | An array of associative arrays, each specifying an origin server for your service. The `port`, `ssl`, and `rules` elements for each origin server are optional. | Array of associative arrays | Yes | `null` | `array( array( 'origin' => 'origin.acme.com', 'port' => 80, 'ssl' => false, 'rules' => array() ) )` |
+| `flavorId` | The flavor used to configure this service. Use the [list flavors](#list-flavors) operation to retrieve a list of all available flavors. | String | Yes| `null` | `cdn` |
+| `restrictions` | An array of associative arrays, each specifying a restriction for who can or cannot access content from the CDN cache. | Array of associative arrays | No | `null` | `array( array( 'name' => 'website only', 'rules' => array( array( 'name' => 'mywebsite.com', 'httpHost' => 'www.mywebsite.com' ) ) ) )` |
+| `caching` | An array of associative arrays, each specifying a caching rule for your service's assets. | No | `null` | `array( array( 'name' => 'default', 'ttl' => 3600 ) )` |
+
+You can update a service as shown in the following example:
+
+```php
+$service->update(array(
+    'origins' => array(
+        array(
+            'origin' => '44.33.22.11',
+            'port'   => 80,
+            'ssl'    => false
+        )
+    )
+));
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/update-service.php) ]
+
+### Delete a service
+
+You can delete a service as shown in the following example:
+
+```php
+$service->delete();
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/delete-service.php) ]
+
+## Service Assets
+A service will have its assets distributed and cached across a CDN's edge nodes.
+
+### Purge all cached service assets
+
+You can purge all cached assets of a service as shown in the following example:
+
+```php
+$service->purgeAssets();
```

Comment:
If a user requests a CDN URL for an assert _after_ this operation, does the API hit the origin? Or does it return a 404?